ecs-cli/modules/cli/servicediscovery/servicediscovery_app.go

```diff
@@ -194,13 +205,22 @@ func create(c *cli.Context, networkMode, serviceName string, cfnClient cloudform
 	return serviceRegistry, err
 }
 
+// createNamespace creates a private DNS namespace
+// This function is used if getExistingNamespace() fails to find an existing namespace with the required settings
+// If a CFN stack with the same name exists already, we therefore know that it doesn't contain a namespace (stack create failed),
+// So we can safely delete the existing stack
 func createNamespace(c *cli.Context, networkMode, serviceName, clusterName string, cfnClient cloudformation.CloudformationClient, input *utils.ServiceDiscovery) (*string, error) {
 	namespaceParams := getNamespaceCFNParams(input)
 	if err := namespaceParams.Validate(); err != nil {
 		return nil, err
 	}
 
 	namespaceStackName := cfnStackName(privateDNSNamespaceStackNameFormat, clusterName, serviceName)
+
+	if err := deleteStack(namespaceStackName, serviceName, "Private DNS Namespace", cfnClient, true, true); err != nil {
+		return nil, errors.Wrapf(err, "A Private DNS Namespace CloudFormation stack for %s already exists, failed to delete existing stack: %s", serviceName, err)
```

Comment:
Is it possible an error could be returned here that is not related to an existing stack? (e.g., network issue, throttling) If so we should make the language more generic.

Reply:
Any error returned here would have to do with the the attempted deletion of the existing stack. With the given arguments, deleteStack will only return an error if the validateStack() call succeeds first. So if there is an error here, we know that the stack exists. Errors could be returned that have to do with throttling or network issues, but since we know that a stack exists, I think that the error message is accurate.

Reply:
While I see what you're trying to do, I agree with Allie that this could be clearer -- having to pass in an unnamed boolean that determines whether validation succeeded requires the reader to look up the implementation details of `deleteStack`(see previous comment).

Reply:
This addresses my concern re: inaccurate messaging, but I concur with @SoManyHs that understanding how we got here is more complicated than ideal (though not a blocker for me ATM)